### DISCUSSION TITLE
Could not poll the database error on docker compose, container "unhealthy"

### DISCUSSION THREAD

**Discussion by WaffleMaster22** [firefly-iii/firefly-iii#8346] [Firefly III]
Im using docker compose running on ubuntu desktop. I tried looking this up and looking through the github but no dice

This is the "Whoops!" page error

Could not poll the database: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for db failed: Name or service not known (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) 

Portainer says the "firefly_iii_core" container is unhealthy but the other containers are fine. This is probably why firefly cant get to the database but idk how to get it to be healthy

This is the logs, this section keeps repeating
https://pastebin.com/Pw4NKnn8

If I leave it running for a little longer then the logs fill with this
https://pastebin.com/KABLeFaA

**Comment by JC5** [comment]
The error seems obvious enough. Is the database running correctly?

**Reply by WaffleMaster22** [reply]
No, it keeps saying "starting" and that's it. When trying to access it I get an error to wait for it to finish starting.

**Reply by JC5** [reply]
What do the logs of the database container say?

**Reply by WaffleMaster22** [reply]
The logs are in the first pastebin. Its kinda hard to read in a pastebin but on the left it says "firefly_iii_db"

**Comment by JC5** [comment]
Ah I see it. It looks like your database is an earlier version of mariadb so it tries to upgrade but fails. How long have you had this install?

**Reply by WaffleMaster22** [reply]
Only the day that I posted the problem. Although ive messed with things that needed a database before.

Also I made things worse... I tried fixing it by re-doing everything and now I cant pull any new images haha
https://www.reddit.com/r/docker/comments/18yl4tx/cant_pull_new_images/
